geopandas/geodataframe.py

```diff
@@ -449,10 +470,19 @@ def crs(self, value):
         else:
             if hasattr(self.geometry.values, "crs"):
                 self.geometry.values.crs = value
-                self._crs = self.geometry.values.crs
             else:
                 # column called 'geometry' without geometry
-                self._crs = None if not value else CRS.from_user_input(value)
+                import pyproj
+
+                self._crs = pyproj.CRS.from_user_input(value)
+
+                # TODO: raise this error in 0.12. This already raises a FutureWarning
```

Comment:
Yes, _column called 'geometry' without geometry_ as the comment on l. 475 says. For now we store CRS on the GeoDataFrame level only in that case but in future, when CRS needs to be linked to geometry this will raise.